.github/workflows/ci_rust.yml

```diff
@@ -213,6 +213,47 @@ jobs:
         run: |
           cargo test --tests --all-features
 
+  # Run the rust unit tests under address sanitizer.
+  #
+  # Rust is generally memory safe, but our bindings contain a large amount of unsafe
+  # code. Additionally, "safe" code doesn't guarentee that the code is free of 
+  # memory leaks.
+  asan-unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        id: toolchain
+        run: |
+          rustup toolchain install ${{env.RUST_NIGHTLY_TOOLCHAIN }} \
+            --profile minimal \
+            --component rust-src \
+            --target x86_64-unknown-linux-gnu
+          rustup override set ${{ env.RUST_NIGHTLY_TOOLCHAIN }}
+      
+      - name: Generate
+        run: ./${{env.ROOT_PATH}}/generate.sh --skip-tests
+
+      # asan expects a binary at /usr/bin/llvm-symbolizer but GHA runners include
+      # multiple versioned binaries, like /usr/bin/llvm-symbolizer-13. This step
+      # finds the latest symbolizer and use it as the "base" llvm-symbolizer binary.
+      #
+      # llvm-symbolizer is necessary to get nice stack traces from asan errors. 
+      # Otherwise the stack trace just contains a hex address like "0x55bc6a28a9b6"
+      - name: set llvm symbolizer
+        run: |
+          sudo ln -s $(find /usr/bin/ -maxdepth 1 -name "llvm-symbolizer-*" | sort -V | tail -n 1) /usr/bin/llvm-symbolizer
+
+      - name: Run Unit Tests under ASAN
+        run: |
+          RUSTDOCFLAGS=-Zsanitizer=address \
```

Comment:
Is there a reason we don't use the `env` fields here?

Reply:
My [original attempt](https://github.com/aws/s2n-tls/pull/4948/commits/89f6c600cc81fc01d52ae3e5d27175e9c873427d) used `env`, but that resulted in failures with the generate step.

```
 Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names -Zsanitizer=address --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: the option `Z` is only accepted on the nightly compiler
```

I believe that is happening because 
1. the override to "nightly" sets a directory-local override
2. the generate script `cd`s into different directories
3. cargo commands in those directories fall back to the version specified in `rust-toolchain`, 1.63 (not nightly)
4. non-nightly toolchains fail to use `-Z` arguments.

Reply:
Ah right. But you can set them just on that run step, rather than the whole thing